client/components/forms/form-radio-with-thumbnail/index.jsx

```diff
@@ -0,0 +1,44 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import FormRadio from 'components/forms/form-radio';
+import FormLabel from 'components/forms/form-label';
+import cssSafeUrl from 'lib/css-safe-url';
+
+const FormRadioWithThumbnail = ( { label, thumbnail, ...otherProps } ) => {
+	const { cssClass, cssColor, imageUrl } = thumbnail;
+	const styles = {
+		backgroundColor: cssColor,
+		backgroundImage: imageUrl ? 'url(' + cssSafeUrl( imageUrl ) + ')' : '',
+	};
```

Comment:
> Components should be flexible and reusable, but always tailored to specific, existing use cases, and be as clear and as simple as possible for that, until a new user case surfaces.

I'd argue that this is exactly what the current solution provides: a simple, yet visually pleasing, solution for the current use case while at the same time being flexible/reusable enough to support alternative use cases.

> We already have patterns for that (see media) [...]
> [...] without any indirection in either CSS nor a style rule.
> [...] we'd also avoid the tricky use of style

Assuming you're referring to `MediaLibraryListItemImage`, the solution there makes use of inline styles  as well (`style={ this.getImageStyle() }`). Said solution involves imitating the behaviour of `object-fit: cover` via JS and inline styles.

>  It's one of these cases where semantic precision ensures simpler code as well as accessible code.

As outlined above for the result we're trying to achieve (reusable, visually appealing, performant, cross-browser) the proposed solution is the simplest solution code wise. 

If being accessible is our prime concern, then using an `<img />` is potentially more accessible.

Again, like stated above, I'm fine with changing the approach if that is what is preferred. I'm therefore going to push up an alternative approach soon. As per your suggestion, it will make use of an `<img />` tag with the current use case as the primary focus.